cluster-autoscaler/simulator/clustersnapshot/clustersnapshot_test.go

```diff
@@ -197,8 +214,34 @@ func TestForking(t *testing.T) {
 				// Modifications should no longer be applied.
 				compareStates(t, tc.state, getSnapshotState(t, snapshot))
 			})
+			t.Run(fmt.Sprintf("%s: %s fork & fork & revert & revert", name, tc.name), func(t *testing.T) {
```

Comment:
I think there should be more test cases like this: fork & fork & commit & revert, fork & fork & revert & commit, with snapshot changes between these steps.

Reply:
Done